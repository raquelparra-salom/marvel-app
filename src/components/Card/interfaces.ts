import { Character } from '../../views/CharactersFavList/interfaces'

export interface ICardProp {
    character: Character
}