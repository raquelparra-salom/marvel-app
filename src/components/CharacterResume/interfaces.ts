import { Character } from '../../views/CharacterList/interfaces';

export interface ICharacterResumeProp {
  character: Character;
}
