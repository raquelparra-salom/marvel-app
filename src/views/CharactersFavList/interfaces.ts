import { Character, Data } from '../CharacterList/interfaces';

export interface CharactersListResponse {
  code: number;
  status: string;
  copyright: string;
  attributionText: string;
  attributionHTML: string;
  etag: string;
  data: Data;
}

export interface Thumbnail {
  path: string;
  extension: string;
}

export interface Comics {
  available: number;
  collectionURI: string;
  items: Item[];
  returned: number;
}

export interface Item {
  resourceURI: string;
  name: string;
}

export interface Series {
  available: number;
  collectionURI: string;
  items: Item2[];
  returned: number;
}

export interface Item2 {
  resourceURI: string;
  name: string;
}

export interface Stories {
  available: number;
  collectionURI: string;
  items: Item3[];
  returned: number;
}

export interface Item3 {
  resourceURI: string;
  name: string;
  type: string;
}

export interface Events {
  available: number;
  collectionURI: string;
  items: Item4[];
  returned: number;
}

export interface Item4 {
  resourceURI: string;
  name: string;
}

export interface IUseCharacterList {
  states: {
    charactersFav: Character[];
    charactersFiltered: Character[];
    filterText: string;
  };
  actions: {
    handleChangeInput: (text: string) => void;
  };
}
