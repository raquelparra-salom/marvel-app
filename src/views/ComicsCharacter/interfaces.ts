import { Character } from '../CharacterList/interfaces';

export interface ComicsCharacterResponse {
  code: number;
  status: string;
  copyright: string;
  attributionText: string;
  attributionHTML: string;
  etag: string;
  data: Data;
}

export interface Data {
  offset: number;
  limit: number;
  total: number;
  count: number;
  results: Result[];
}

export interface Result {
  id: number;
  digitalId: number;
  title: string;
  issueNumber: number;
  variantDescription: string;
  description?: string;
  modified: string;
  isbn: string;
  upc: string;
  diamondCode: string;
  ean: string;
  issn: string;
  format: string;
  pageCount: number;
  textObjects: TextObject[];
  resourceURI: string;
  urls: Url[];
  series: Series;
  variants: Variant[];
  collections: string[];
  collectedIssues: string[];
  dates: Date[];
  prices: Price[];
  thumbnail: Thumbnail;
  images: Image[];
  creators: Creators;
  characters: Characters;
  stories: Stories;
  events: Events;
}

export interface TextObject {
  type: string;
  language: string;
  text: string;
}

export interface Url {
  type: string;
  url: string;
}

export interface Series {
  resourceURI: string;
  name: string;
}

export interface Variant {
  resourceURI: string;
  name: string;
}

export interface Date {
  type: string;
  date: string;
}

export interface Price {
  type: string;
  price: number;
}

export interface Thumbnail {
  path: string;
  extension: string;
}

export interface Image {
  path: string;
  extension: string;
}

export interface Creators {
  available: number;
  collectionURI: string;
  items: Item[];
  returned: number;
}

export interface Item {
  resourceURI: string;
  name: string;
  role: string;
}

export interface Characters {
  available: number;
  collectionURI: string;
  items: {
    resourceURI: string;
    name: string;
  }[];
  returned: number;
}

export interface Stories {
  available: number;
  collectionURI: string;
  items: {
    resourceURI: string;
    name: string;
    type: string;
  }[];
  returned: number;
}

export interface Events {
  available: number;
  collectionURI: string;
  items: {
    resourceURI: string;
    name: string;
  }[];
  returned: number;
}

export interface IUseComicsCharacter {
  states: {
    comics?: ComicsCharacterResponse;
    character?: Character;
    loading: boolean;
  };
}
