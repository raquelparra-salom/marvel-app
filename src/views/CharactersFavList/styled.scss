


.cards-container {
  --auto-grid-min-size: 14rem;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(var(--auto-grid-min-size), 1fr));
  gap: 20px;
  padding: 20px;
}

