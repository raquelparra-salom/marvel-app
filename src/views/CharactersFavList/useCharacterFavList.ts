import { useState, useContext, useEffect, useMemo } from 'react';
import { IUseCharacterList } from './interfaces';
import './styled.scss';
import { FavoriteContext } from '../../components/Root/Root';
import { Character } from '../CharacterList/interfaces';

const useCharacterFavList = (): IUseCharacterList => {
  const { charactersFav } = useContext(FavoriteContext);
  const [filterText, setFilterText] = useState<string>('');

  const charactersFiltered = useMemo(() => {
    if (!filterText) {
      return charactersFav;
    }
    return charactersFav.filter((character) =>
      character.name.toLocaleLowerCase().includes(filterText)
    );
  }, [charactersFav, filterText]);

  const handleChangeInput = (text: string) => {
    setFilterText(text);

  };
  return {
    states: { charactersFav, charactersFiltered, filterText },
    actions: { handleChangeInput},
  };
};

export default useCharacterFavList;
