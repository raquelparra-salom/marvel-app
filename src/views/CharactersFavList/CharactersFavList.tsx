import { Card } from '../../components/Card/';
import './styled.scss';
import { Filter } from '../../components/Filter/';
import useCharacterFavList from './useCharacterFavList';

export function CharactersFavList() {
  const {
    states: { charactersFiltered, filterText },
    actions: { handleChangeInput },
  } = useCharacterFavList();

  return (
    <>
      <Filter handleChange={handleChangeInput} value={filterText} count={charactersFiltered.length} />
      <div className="cards-container">
        {charactersFiltered.map((character) => (
          <Card key={character.id} character={character} />
        ))}
      </div>
    </>
  );
}
