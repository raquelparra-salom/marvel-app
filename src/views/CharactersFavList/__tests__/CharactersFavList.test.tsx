import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { CharactersFavList } from '../CharactersFavList';
import { FavoriteContext } from '../../../components/Root/Root';
import { IFavoriteContext } from '../../../components/Root/interfaces';
import { character1, character2 } from './data';

describe('CharactersFavList component', () => {
  it('Should render CharactersFavList component ', async () => {

    const context: IFavoriteContext = {
      charactersFav: [character1, character2],
      addFav: jest.fn(),
      deleteFav: jest.fn()
    }

    const { container } = render(
      <MemoryRouter>
        <FavoriteContext.Provider value={context}>
          <CharactersFavList />
        </FavoriteContext.Provider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
