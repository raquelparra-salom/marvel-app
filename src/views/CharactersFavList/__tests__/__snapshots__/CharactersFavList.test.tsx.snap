// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersFavList component Should render CharactersFavList component  1`] = `
<div>
  <div
    class="filter-wrapper"
  >
    <div
      class="searcher"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass "
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
      <input
        placeholder="Search a Character..."
        type="text"
        value=""
      />
    </div>
    <div
      class="counter"
    >
      2
       Results
    </div>
  </div>
  <div
    class="cards-container"
  >
    <div
      class="card"
    >
      <a
        href="/comics/1"
      >
        <img
          class="image"
          src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
        />
      </a>
      <div
        class="info"
      >
        <span
          class="title"
        >
          3-D Man 1
        </span>
        <div
          class="favorite-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart heart-selected"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <a
        href="/comics/2"
      >
        <img
          class="image"
          src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
        />
      </a>
      <div
        class="info"
      >
        <span
          class="title"
        >
          3-D Man 2
        </span>
        <div
          class="favorite-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart heart-selected"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
